### PR TITLE
gh-122292: Split up ``Lib/test/test_ast.py`` into a couple of files

### PR DESCRIPTION
**Pull request by Eclips4**
* Issue: gh-122292

### CODE REVIEW

**Review by sobolevn — APPROVED** [review]
I like the idea of simplier tests. Backports are required, so we can manage this with ease in the future.

File: Lib/test/support/ast_helper.py [hunk 1]
```diff
@@ -41,3 +41,19 @@ def traverse_compare(a, b, missing=object()):
             elif a != b:
                 self.fail(f"{a!r} != {b!r}")
         traverse_compare(ast1, ast2)
+
+def to_tuple(t):
```
Comment (vstinner) [inline Lib/test/support/ast_helper.py]: Would it be possible to add it to `Lib/test/test_ast/__init__.py` or (new file) Lib/test/test_ast/utils.py?

File: Lib/test/test_ast/test_snippets.py [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+            print(kind+"_results = [")
+            for statement in statements:
+                tree = ast.parse(statement, "?", kind)
+                print("%r," % (to_tuple(tree),))
+            print("]")
+        print("main()")
+        raise SystemExit
+    unittest.main()
+
+#### EVERYTHING BELOW IS GENERATED BY python Lib/test/test_ast/test_cases.py -g  #####
```
Comment (vstinner) [inline Lib/test/test_ast/test_snippets.py]: Would it be possible to put generated data in a separated file?

Comment (Eclips4) [reply Lib/test/test_ast/test_snippets.py]: I've decided to move the snippets and generated data to `test_ast/snippets.py`. I think it is better, so there is no need for an extra file `test_snippets.py`

File: Lib/test/test_ast/__init__.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import os
+import unittest
```
Comment (vstinner) [inline Lib/test/test_ast/__init__.py]: this import is not used

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by vstinner** [comment]
@Eclips4: It seems like the backport should be done manually. Can you do it for 3.13?

**Comment by Eclips4** [comment]
> @Eclips4: It seems like the backport should be done manually. Can you do it for 3.13?

Yes, I would! I'd also like to backport 3.12. Or do you think we should not backport this change to 3.12?

**Pull request merged**